### ISSUE TITLE
How to use sample_weight when we have TimeDistributed layers?

### ISSUE THREAD

**Issue by amirveyseh** [fchollet/keras#7497]
I've the following sequential model:

model = Sequential()     model.add(TimeDistributed(Embedding(_VOCABULARY_SIZE+2,_EMBEDDING_SIZE,mask_zero=True,input_length=_MAX_TWEET_LENGTH), input_shape=(_MAX_TWEET_SEQ_LENGTH, _MAX_TWEET_LENGTH)))
        model.add(TimeDistributed(LSTM(128, dropout=0.2, recurrent_dropout=0.2, input_shape=(_MAX_TWEET_LENGTH, _EMBEDDING_SIZE))))
        model.add(Dense(10, activation='softmax'))
        model.add(LSTM(128, dropout=0.2, recurrent_dropout=0.2))
        model.add(Dense(3, activation='softmax'))
        model.compile(loss='categorical_crossentropy', optimizer='RMSprop', metrics=['accuracy'], sample_weight_mode='temporal')
        model.fit(np.asarray(x_train), np.asarray(y_train), batch_size=_BATCH_SIZE, epochs=_EPOCH, validation_split=0.2, sample_weight=sample_weights)

The input_shape is : (18, 2000, 37)
and the output_shape is: (18, 3)

in each of these 18 samples (which are indeed sequence of tweets) I want to ignore some of the tweets. I've calculated a sample_weight of shape (18, 2000) to indicate which ones should not contribute to the final loss function. But I get this error: 

ValueError: Found a sample_weight array for an input with shape (18, 3). Timestep-wise sample weighting (use of sample_weight_mode="temporal") is restricted to outputs that are at least 3D, i.e. that have a time dimension.

My output could not be temporal but my input is. How can I get around this issue?

Thanks

**Comment by mahnerak** [comment]
`sample_weight` is used as **weights for loss**, but your loss is with shape `(18, 3)`.
All the temporal information is lost at `model.add(Dense(10, activation='softmax'))`, there's no way to controll/add weight to sequence element after that, so what you want to do is a non-sense mathematically.
TL;DR Your model doesn't support sample weighting like that, work on your model.
